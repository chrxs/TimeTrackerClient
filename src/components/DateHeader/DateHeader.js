import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import moment from 'moment'

import styles from './DateHeader.scss'

const DateHeader = ({
  year,
  month,
  day
}) => {
  let interval = 'year'
  let date = year
  let titleFormat = 'YYYY'
  let linkFormat = '/YYYY'

  if (month) {
    interval = 'month'
    linkFormat += '/MM'
    date += `-${month}`
    titleFormat = `MMMM ${titleFormat}`
  } else {
    date += '-01'
  }

  if (day) {
    interval = 'day'
    linkFormat += '/DD'
    date += `-${day}`
    titleFormat = `dddd, Do ${titleFormat}`
  } else {
    date += '-01'
  }

  return (
    <header className={styles.DateHeader}>
      <Link
        to={moment(date).subtract(1, interval).format(linkFormat)}
        className={[styles.Link].join(' ')}
      >
        <i className='fa fa-angle-left' />
      </Link>
      <h1 className={styles.DateHeaderTitle}>
        {moment(date).format(titleFormat)}
      </h1>
      <Link
        to={moment(date).add(1, interval).format(linkFormat)}
        className={[styles.Link].join(' ')}
      >
        <i className='fa fa-angle-right' />
      </Link>
    </header>
  )
}

DateHeader.displayName = 'DateHeader'

DateHeader.propTypes = {
  year: PropTypes.string,
  month: PropTypes.string,
  day: PropTypes.string
}

DateHeader.defaultProps = {
  year: new Date().getFullYear()
}

export default DateHeader
