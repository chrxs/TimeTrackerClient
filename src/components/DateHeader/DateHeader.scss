.DateHeader {
  text-align: center;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin: 0;
  padding: 15px;
  height: 40px;
  border-bottom: solid 1px #666;
}

.DateHeaderTitle {
  margin: 0;
  padding: 0;
  font-size: 22px;
  font-weight: 200;
}

.Link {
  background-color: transparent;
  background-size: contain;
  background-repeat: no-repeat;
  background-position: 50% 50%;
  height: 40px;
  width: 40px;
}

.previous {
  background-image: url(data:image/svg+xml;base64,PHN2ZyB2aWV3Qm94PSIyMzkgMjU5IDI1IDQxIiB2ZXJzaW9uPSIxLjEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiPiAgPHBhdGggZD0iTTI2My4xOTE5OTQsMjYzLjE0MjgxIEMyNjMuMTkxOTk0LDI2Mi43ODEyMDIgMjYzLjAzMTI4LDI2Mi40NTk3NzMgMjYyLjc5MDIwOCwyNjIuMjE4NzAxIEwyNjAuNzgxMjc3LDI2MC4yMDk3NyBDMjYwLjU0MDIwNSwyNTkuOTY4Njk5IDI2MC4xNzg1OTgsMjU5LjgwNzk4NCAyNTkuODU3MTY5LDI1OS44MDc5ODQgQzI1OS41MzU3NCwyNTkuODA3OTg0IDI1OS4xNzQxMzIsMjU5Ljk2ODY5OSAyNTguOTMzMDYxLDI2MC4yMDk3NyBMMjQwLjIwOTgyNCwyNzguOTMzMDA3IEMyMzkuOTY4NzUyLDI3OS4xNzQwNzkgMjM5LjgwODAzOCwyNzkuNTM1Njg2IDIzOS44MDgwMzgsMjc5Ljg1NzExNSBDMjM5LjgwODAzOCwyODAuMTc4NTQ0IDIzOS45Njg3NTIsMjgwLjU0MDE1MiAyNDAuMjA5ODI0LDI4MC43ODEyMjQgTDI1OC45MzMwNjEsMjk5LjUwNDQ2IEMyNTkuMTc0MTMyLDI5OS43NDU1MzIgMjU5LjUzNTc0LDI5OS45MDYyNDYgMjU5Ljg1NzE2OSwyOTkuOTA2MjQ2IEMyNjAuMTc4NTk4LDI5OS45MDYyNDYgMjYwLjU0MDIwNSwyOTkuNzQ1NTMyIDI2MC43ODEyNzcsMjk5LjUwNDQ2IEwyNjIuNzkwMjA4LDI5Ny40OTU1MjkgQzI2My4wMzEyOCwyOTcuMjU0NDU3IDI2My4xOTE5OTQsMjk2Ljg5Mjg1IDI2My4xOTE5OTQsMjk2LjU3MTQyMSBDMjYzLjE5MTk5NCwyOTYuMjQ5OTkyIDI2My4wMzEyOCwyOTUuODg4Mzg0IDI2Mi43OTAyMDgsMjk1LjY0NzMxMyBMMjQ3LjAwMDAxMSwyNzkuODU3MTE1IEwyNjIuNzkwMjA4LDI2NC4wNjY5MTggQzI2My4wMzEyOCwyNjMuODI1ODQ2IDI2My4xOTE5OTQsMjYzLjQ2NDIzOSAyNjMuMTkxOTk0LDI2My4xNDI4MSBaIiBzdHJva2U9Im5vbmUiIGZpbGw9IiMwMDAwMDAiPjwvcGF0aD48L3N2Zz4=);
}

.next {
  background-image: url(data:image/svg+xml;base64,PHN2ZyB2aWV3Qm94PSIzNDYgMjY1IDI0IDQxIiB2ZXJzaW9uPSIxLjEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiPiAgICA8cGF0aCBkPSJNMzY5LjkwNjI3OCwyODUuODU3MTE1IEMzNjkuOTA2Mjc4LDI4NS41MzU2ODYgMzY5Ljc0NTU2NCwyODUuMTc0MDc5IDM2OS41MDQ0OTIsMjg0LjkzMzAwNyBMMzUwLjc4MTI1NiwyNjYuMjA5NzcgQzM1MC41NDAxODQsMjY1Ljk2ODY5OSAzNTAuMTc4NTc2LDI2NS44MDc5ODQgMzQ5Ljg1NzE0NywyNjUuODA3OTg0IEMzNDkuNTM1NzE5LDI2NS44MDc5ODQgMzQ5LjE3NDExMSwyNjUuOTY4Njk5IDM0OC45MzMwMzksMjY2LjIwOTc3IEwzNDYuOTI0MTA4LDI2OC4yMTg3MDEgQzM0Ni42ODMwMzcsMjY4LjQ1OTc3MyAzNDYuNTIyMzIyLDI2OC44MjEzODEgMzQ2LjUyMjMyMiwyNjkuMTQyODEgQzM0Ni41MjIzMjIsMjY5LjQ2NDIzOSAzNDYuNjgzMDM3LDI2OS44MjU4NDYgMzQ2LjkyNDEwOCwyNzAuMDY2OTE4IEwzNjIuNzE0MzA2LDI4NS44NTcxMTUgTDM0Ni45MjQxMDgsMzAxLjY0NzMxMyBDMzQ2LjY4MzAzNywzMDEuODg4Mzg0IDM0Ni41MjIzMjIsMzAyLjI0OTk5MiAzNDYuNTIyMzIyLDMwMi41NzE0MjEgQzM0Ni41MjIzMjIsMzAyLjkzMzAyOCAzNDYuNjgzMDM3LDMwMy4yNTQ0NTcgMzQ2LjkyNDEwOCwzMDMuNDk1NTI5IEwzNDguOTMzMDM5LDMwNS41MDQ0NiBDMzQ5LjE3NDExMSwzMDUuNzQ1NTMyIDM0OS41MzU3MTksMzA1LjkwNjI0NiAzNDkuODU3MTQ3LDMwNS45MDYyNDYgQzM1MC4xNzg1NzYsMzA1LjkwNjI0NiAzNTAuNTQwMTg0LDMwNS43NDU1MzIgMzUwLjc4MTI1NiwzMDUuNTA0NDYgTDM2OS41MDQ0OTIsMjg2Ljc4MTIyNCBDMzY5Ljc0NTU2NCwyODYuNTQwMTUyIDM2OS45MDYyNzgsMjg2LjE3ODU0NCAzNjkuOTA2Mjc4LDI4NS44NTcxMTUgWiIgc3Ryb2tlPSJub25lIiBmaWxsPSIjMDAwMDAwIj48L3BhdGg+PC9zdmc+);
}
