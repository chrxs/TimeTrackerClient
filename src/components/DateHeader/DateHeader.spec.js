/* global describe, test, expect */
import React from 'react'
import { BrowserRouter } from 'react-router-dom'

import DateHeader from './DateHeader'
import renderer from 'react-test-renderer'

describe('DateHeader snapshots', () => {
  test('with year prop', () => {
    const component = renderer.create(
      <BrowserRouter>
        <DateHeader year='2017' />
      </BrowserRouter>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('with year and month props', () => {
    const component = renderer.create(
      <BrowserRouter>
        <DateHeader year='2017' month='04' />
      </BrowserRouter>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('with year, month and day props', () => {
    const component = renderer.create(
      <BrowserRouter>
        <DateHeader year='2017' month='04' day='21' />
      </BrowserRouter>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
