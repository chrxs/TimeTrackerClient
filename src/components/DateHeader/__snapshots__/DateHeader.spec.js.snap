// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateHeader snapshots with year and month props 1`] = `
<header
  className="DateHeader"
>
  <a
    className="Link previous"
    href="/2017/03"
    onClick={[Function]}
  />
  <h1
    className="DateHeaderTitle"
  >
    April 2017
  </h1>
  <a
    className="Link next"
    href="/2017/05"
    onClick={[Function]}
  />
</header>
`;

exports[`DateHeader snapshots with year prop 1`] = `
<header
  className="DateHeader"
>
  <a
    className="Link previous"
    href="/2016"
    onClick={[Function]}
  />
  <h1
    className="DateHeaderTitle"
  >
    2017
  </h1>
  <a
    className="Link next"
    href="/2018"
    onClick={[Function]}
  />
</header>
`;

exports[`DateHeader snapshots with year, month and day props 1`] = `
<header
  className="DateHeader"
>
  <a
    className="Link previous"
    href="/2017/04/20"
    onClick={[Function]}
  />
  <h1
    className="DateHeaderTitle"
  >
    Friday, 21st April 2017
  </h1>
  <a
    className="Link next"
    href="/2017/04/22"
    onClick={[Function]}
  />
</header>
`;
